flocker/node/agents/blockdevice.py

```diff
@@ -1339,7 +1364,13 @@ def get_device_path(self, blockdevice_id):
              volume.blockdevice_id.encode("ascii")]
         )
         # May be None if the file hasn't been used for a loop device.
-        return _device_for_path(volume_path)
+        path = _device_for_path(volume_path)
+        if path is None:
+            # It was supposed to be attached but it has no loopback device.  So
```

Comment:
Shouldn't calling `get_device_path` on some random block id that has not device be a bug in the caller? In which case this should raise exception.